tools/profiler/profiler-process/construct_trace.py

```diff
@@ -298,51 +299,44 @@ def value(self, vcd, time, value, identifier_code, cur_sig_vals):
                         int_value
                     )
 
-    """
-    Postprocess data mapping timestamps to events (signal changes)
-    We have to postprocess instead of processing signals in a stream because
-    signal changes that happen at the same time as a clock tick might be recorded
-    *before* or *after* the clock change on the VCD file (hence why we can't process
-    everything within a stream if we wanted to be precise)
-    """
-
     def postprocess(self, shared_cells_map):
-        clock_name = f"{self.main_component}.clk"
+        """
+        Postprocess data mapping timestamps to events (signal changes)
+        We have to postprocess instead of processing signals in a stream because
+        signal changes that happen at the same time as a clock tick might be recorded
+        *before* or *after* the clock change on the VCD file (hence why we can't process
+        everything within a stream if we wanted to be precise)
+        """
+
         clock_cycles = -1  # will be 0 on the 0th cycle
         started = False
         cell_active = set()
         group_active = set()
         structural_enable_active = set()
         cell_enable_active = set()
         primitive_enable = set()
-        trace = {}
-        trace_classified = []
-        cell_to_active_cycles_summary = {}  # cell --> {"num-times-active": _, "active-cycles": []}
-        # we lose information about the length of each segment but we can retrieve that information from the timeline
 
         # The events are "partial" because we don't know yet what the tid and pid would be.
         # (Will be filled in during create_timelines(); specifically in port_fsm_events())
-        fsm_current = {fsm: 0 for fsm in self.fsms}  # fsm --> value
+        fsm_current = {fsm: 0 for fsm in self.control_metadata.fsms}  # fsm --> value
 
         probe_labels_to_sets = {
             "group_probe_out": group_active,
             "se_probe_out": structural_enable_active,
             "cell_probe_out": cell_enable_active,
             "primitive_probe_out": primitive_enable,
         }
-
         main_done = False  # Prevent creating a trace entry for the cycle where main.done is set high.
         for ts in self.timestamps_to_events:
             events = self.timestamps_to_events[ts]
-            started = started or [
-                x
-                for x in events
-                if x["signal"] == f"{self.main_component}.go" and x["value"] == 1
-            ]
+            started = (
+                started
+                or WaveformEvent(f"{self.cell_metadata.main_component}.go", 1) in events
```

Comment:
just wanted to flag that `events` is a list so this `in` check is doing a linear
scan